packages/adapter-components/test/elements/swagger/instance_elements.test.ts

```
@@ -1029,5 +1030,136 @@ describe('swagger_instance_elements', () => {
+          },
+        })
+        return (await getAllInstances({
+          paginator,
+          apiConfig: {
+            typeDefaults: {
+              transformation: {
+                idFields: ['id'],
+              },
+            },
+            types: {
+              customType: {
+                request: {
+                  url: '/someObject',
+                },
+                transformation: {
+                  fileNameFields,
+                  idFields,
+                },
+              },
+            },
+          },
+          fetchConfig: {
+            includeTypes: ['customType'],
+          },
+          objectTypes: { customType },
+          computeGetArgs: simpleGetArgs,
+          nestedFieldFinder: returnFullEntry,
+        }))[0]
+      }
+
+      describe('element id', () => {
+        const getInstanceIdName = async (params: GetInstanceParams) : Promise<string> => {
+          const { elemID } = await getInstance(params)
+          return elemID.name
+        }
+        /**
+         * This test demonstrates a bug that causes collusion between element ids.
+         * Please refer to [this]{@link https://salto-io.atlassian.net/browse/SALTO-1738} ticket
+         */
+        it('generates id with value "null" when one null field is given', async () => {
+          const elementIdName = await getInstanceIdName({ idFields: ['nullField'] })
+          expect(elementIdName).toEqual(naclCase('null'))
+        })
+        it('generates id with non undefined fields', async () => {
+          const elementIdName = await getInstanceIdName({ idFields: ['id', 'name', 'number', 'list', 'nullField'] })
+          expect(elementIdName).toEqual(naclCase(`${ID}_${NAME}_${NUMBER}_${LIST}_null`))
+        })
+
+        describe('uses default name as id', () => {
+          const EXPECTED_DEFAULT_NAME = 'unnamed_0'
+          it('no fields are provided', async () => {
+            const elementIdName = await getInstanceIdName({ idFields: [] })
```

as mentioned above, this should not be a valid scenario so we should throw an error instead